hack/run-functests.sh

```diff
@@ -32,102 +36,77 @@ elif [ "$FEATURES" == "" ]; then
+  FOCUS="--focus="$(echo "$FEATURES" | tr ' ' '|')
   if [ "$FOCUS_TESTS" != "" ]; then
-    FOCUS="-ginkgo.focus="$(echo "$FOCUS_TESTS" | tr ' ' '|')
+    FOCUS="--focus="$(echo "$FOCUS_TESTS" | tr ' ' '|')
   fi
   echo "Focusing on $FOCUS"
 fi
 
 if [ "$SKIP_TESTS" != "" ]; then
-	SKIP="-ginkgo.skip="$(echo "$SKIP_TESTS" | tr ' ' '|')
+	SKIP="--skip="$(echo "$SKIP_TESTS" | tr ' ' '|')
 	echo "Skip set, skipping $SKIP"
 fi
 
 export SUITES_PATH=cnf-tests/bin
 
-mkdir -p "$TESTS_REPORTS_PATH"
-
-
-
-if [ "$TESTS_IN_CONTAINER" == "true" ]; then
-  cp -f "$KUBECONFIG" _cache/kubeconfig
-  echo "Running dockerized version via $TEST_EXECUTION_IMAGE"
-
-  features="$(echo "$FEATURES" | tr ' ' '|')"
-
-  env_vars="-e CLEAN_PERFORMANCE_PROFILE=false \
-  -e CNF_TESTS_IMAGE=$TEST_POD_CNF_TEST_IMAGE \
-  -e DPDK_TESTS_IMAGE=$TEST_POD_DPDK_TEST_IMAGE \
-  -e IMAGE_REGISTRY=$TEST_POD_IMAGES_REGISTRY \
-  -e KUBECONFIG=/kubeconfig/kubeconfig \
-  -e SCTPTEST_HAS_NON_CNF_WORKERS=$SCTPTEST_HAS_NON_CNF_WORKERS \
-  -e TEST_SUITES=$TEST_SUITES \
-  -e IS_OPENSHIFT=$IS_OPENSHIFT \
-  -e FEATURES=$features \
-  -e OO_INSTALL_NAMESPACE=$OO_INSTALL_NAMESPACE"
-
-  # add latency tests env variable to the cnf-tests container
-  if [ "$LATENCY_TEST_RUN" == "true" ];then
-    env_vars="$env_vars \
-    -e LATENCY_TEST_RUN=$LATENCY_TEST_RUN \
-    -e LATENCY_TEST_RUNTIME=$LATENCY_TEST_RUNTIME \
-    -e LATENCY_TEST_DELAY=$LATENCY_TEST_DELAY \
-    -e OSLAT_MAXIMUM_LATENCY=$OSLAT_MAXIMUM_LATENCY \
-    -e CYCLICTEST_MAXIMUM_LATENCY=$CYCLICTEST_MAXIMUM_LATENCY \
-    -e HWLATDETECT_MAXIMUM_LATENCY=$HWLATDETECT_MAXIMUM_LATENCY \
-    -e MAXIMUM_LATENCY=$MAXIMUM_LATENCY"
+mkdir -pv "$TESTS_REPORTS_PATH/junit"
+mkdir -pv "$TESTS_REPORTS_PATH/raw"
+go build -o cnf-tests/bin/junit-merger cnf-tests/testsuites/pkg/junit-merger/junit-merger.go
+
+TEST_SUITES=${TEST_SUITES:-"validationsuite configsuite cnftests"}
+suites=( $TEST_SUITES )
+
+for suite in "${suites[@]}"; do
+# If the CNF_COMPONENTS variable is empty, run all the components for suite
+  if [[ -z "$CNF_COMPONENTS" ]]; then
+    case $suite in
+      "configsuite")
+        suite_components=("nto")
+        ;;
+      "validationsuite")
+        suite_components=("cluster" "metallb")
+        ;;
+      "cnftests")
+        suite_components=("integration" "metallb" "nto-performance" "nto-latency" "ptp" "sriov")
+        ;;
+      *)
+        echo "Invalid suite name: $suite"
+        exit 1
+        ;;
+    esac
+  else
+    suite_components=($CNF_COMPONENTS)
   fi
 
-  EXEC_TESTS="$CONTAINER_MGMT_CLI run \
-  -v $(pwd)/_cache/:/kubeconfig:Z \
-  -v $TESTS_REPORTS_PATH:/reports:Z \
-  --network host \
-  ${env_vars} \
-  $TEST_EXECUTION_IMAGE /usr/bin/test-run.sh $FAIL_FAST $SKIP $FOCUS $GINKGO_PARAMS -junit /reports/ -report /reports/"
-else
-  cnf-tests/hack/build-test-bin.sh
-  EXEC_TESTS="cnf-tests/entrypoint/test-run.sh $FAIL_FAST $SKIP $FOCUS $GINKGO_PARAMS -junit $TESTS_REPORTS_PATH -report $TESTS_REPORTS_PATH"
```

Comment:
Without this `-report` parameter, we won't be able to debug failing tests. Is there any issue with keeping it?

Reply:
Deleted it accidentally when I moved to the --juint-report flag. thanks

Reply:
Hey @zeeke just an update on that convo, I removed it temporarily for the ci to pass, because the external test suites don't have this flag yet.